harness/determined/common/api/certs.py

```diff
@@ -0,0 +1,220 @@
+import atexit
+import contextlib
+import json
+import logging
+import os
+import pathlib
+import tempfile
+from typing import Dict, Iterator, Optional, Union, cast
+
+import certifi
+import filelock
+
+from determined.common.api import authentication
+
+
+class Cert:
+    def __init__(
+        self,
+        cert_pem: Optional[str] = None,
+        noverify: bool = False,
+        name: Optional[str] = None,
+    ) -> None:
+        if cert_pem is not None and noverify:
+            raise AssertionError("you cannot set cert_pem with noverify=True")
+
+        if name == "":
+            name = None
+        self._name = name
+
+        if noverify:
+            self._bundle = False  # type: Union[None, str, bool]
+        elif cert_pem is None:
+            self._bundle = None
+        else:
+            # Don't use NamedTemporaryFile, since it would make the file inaccessible by path on
+            # Windows after this.
+            # (see https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile)
+            fd, combined_path = tempfile.mkstemp(prefix="det-master-cert-")
+            atexit.register(os.unlink, combined_path)
+
+            with open(fd, "wb") as out:
+                with open(certifi.where(), "rb") as base_certs:
+                    out.write(base_certs.read())
+                out.write(b"\n")
+                out.write(cert_pem.encode("utf8"))
+
+            self._bundle = combined_path
+
+    @property
+    def bundle(self) -> Union[None, str, bool]:
+        """
+        The path to a file containing an SSL certificate to trust specifically for the master, if
+        any, or False to disable cert verification entirely. If set to a path, it should always be a
+        temporary file that we own and can delete.
+        """
+        return self._bundle
+
+    @property
+    def name(self) -> Optional[str]:
+        """
+        The name we use to verify the master certificate.
+        """
+        return self._name
+
+
+cli_cert = None  # type: Optional[Cert]
+
+
+class _BrokenCertStore(Exception):
+    pass
```

Comment:
Add a doc string to this to explain which would cause this to happen

Reply:
Refactoring this to have the same behavior as the token store; I like that behavior much better.

Also renaming to `api.errors.CorruptCertificateCacheException`, which I think is probably clearer.